docs/shim-rune.md

```diff
@@ -0,0 +1,243 @@
+# Design
+
+According to [Enclave Confidential Containers (Enclave-CC) Design and Architecture](./design.md),
+shim-rune should launch the agent enclave container when pod creation. The agent enclave container
+is deployed in the form of OCI bundle (pre-installed on each node) rather than a container image,
+and thus `shim-rune` can call `rune` to start it. The agent enclave container has the same life
+cycle as the pod. Its main job is to receive and then load the actual workload's container image
+into an app enclave.
+
+![image](./images/shim.png)
```

Comment:
What do you want to communicate with the arrow from PAL to rune?
I think the other arrows loosely suggest call flow.

Reply:
Hi, thank you for your advice.
I have updated the [shim.png](https://github.com/haosanzi/enclave-cc/blob/main/docs/images/shim.png) to show the relationship with rune and pal/pal api.

PAL is the implementer of Enclave Runtime PAL API. [Enclave Runtime PAL API](https://github.com/inclavare-containers/inclavare-containers/blob/master/rune/libenclave/internal/runtime/pal/spec.md) defines a common interface to interact between rune and enclave runtime. `rune` manages the confidential container through the PAL API.

- [PAL API](https://github.com/inclavare-containers/inclavare-containers/blob/master/rune/libenclave/internal/runtime/pal/spec.md): This API defines the function calls between Enclave Runtime PAL and rune. 
- PAL: The implementer of Enclave Runtime PAL API, on behalf of Enclave Runtime. This [repo](https://github.com/occlum/occlum/tree/master/src/pal)  shows the implementation of occlum pal.
- Enclave Runtime: The implementer of enclave. Occlum, Graphene-SGX, and WAMR (WebAssembly Micro Runtime) are all the so-called Enclave Runtime. 

Please refer to [guide](https://github.com/inclavare-containers/inclavare-containers/blob/master/docs/design/terminology.md) for the terminology in inclavare-container.